README.md

```diff
@@ -1,634 +1,1828 @@
+Capital-at-rest is wasted economic potential. A high TVL fundamentally means the supply is more than
+the demand. The higher the TVL, the greater the disconnect between the supply and demand. This is a
+huge misallocation of resources. If the DeFi economy is going to reach its full potential, **TVL
+should be as low as possible.**
+
+### No Trustless Composability with other DApps
+
+DApp protocols that require users to go through batchers will never be as trustlessly composable as
+fully p2p protocols. The argument was already made in the cardano-swaps
+[README](https://github.com/fallen-icarus/cardano-swaps#why-the-workarounds-for-the-issues-of-the-lp-based-architecture-are-not-sufficient)
+so it will just be quoted here:
+
+> - Batcher based protocols cannot trustlessly compose - since LPs require going through middle-men,
+the ultimate transaction for Alice will likely not even be seen by Alice prior to submission to the
+blockchain. How can Alice express that she wants her swap composed with an options contract and
+expect it to be trustlessly executed? How can Bob also do this? Finally, how can the smart contracts
+enforce that *both* Alice's and Bob's compositions are properly executed when their requests are
+batched together? This is not an issue of standards; this is an issue of whether or not the required
+composition logic can even fit in a transaction where other DApp logic is being executed. The only
+way for the composition to be trustless is if smart contracts enforce it. To put it concretely, in
+Alice's composition, the swap logic and the options logic must both be executed in the same
+transaction where the composition logic is executed. Given the very small amount of execution units
+available to scripts in a transaction, is there even room for this extra logic? The composition
+logic requires two things: it must support *arbitrary* compositions and it must be extremely cheap
+since the logic of the actual DApps being composed must also be executed in the same transaction.
+*p2p protocols do not have the same limitations since Alice personally creates and signs her own
+transaction.* The above two requirements are only requirements when middle-men are creating and
+executing transactions on the behalf of users. Even if some composition logic could fit in the
+transaction, would it actually support arbitrary composition (as opposed to only a few priviledged
+DApps being supported)? Even if the answer to this question is yes, that extra logic being executed
+means less space for other DApps to compose in the transaction. Therefore, protocols that require
+batchers will never be able to offer the the same level of trustless composition as fully p2p
+protocols. The only way to actually match the level of composition of p2p protocols is for the
+batcher based composition to *not be trustless* (without the composition logic, more DApps can fit
+in a transaction).
```

Comment:
Even if not broken up into paragraphs in the original, it should be broken up here... not readable since requires too many details be accumulated in the reader's head between moving from one conceptual step to another.